crates/python/src/qpu/client.rs

```diff
@@ -40,18 +51,150 @@ py_wrap_error!(
     PyRuntimeError
 );
 
+wrap_error!(ConfigurationBuildError(BuildError));
+py_wrap_error!(
+    qcs,
+    ConfigurationBuildError,
+    QcsConfigurationBuildError,
+    PyRuntimeError
+);
+
+fn get_pydict_str(py_dict: &PyDict, key: &str) -> Option<String> {
+    py_dict.get_item(key)?.extract().ok()
+}
+
+py_wrap_struct! {
+    PyQcsClientAuthServer(AuthServer) as "QcsClientAuthServer" {
+        py -> rs {
+            py_dict: Py<PyDict> => AuthServer {
+                let py_dict = py_dict.as_ref(py);
+                let mut auth_server = AuthServer::default();
+                if let Some(client_id) = get_pydict_str(py_dict, "client_id") {
+                    auth_server = auth_server.set_client_id(client_id);
+                }
+                if let Some(issuer) = get_pydict_str(py_dict, "issuer") {
+                    auth_server = auth_server.set_issuer(issuer);
+                }
+                Ok::<_, PyErr>(auth_server)
+            }
+        },
+        rs -> py {
+            rs_struct: AuthServer => Py<PyDict> {
+                let obj = PyDict::new(py);
+                obj.set_item("client_id", rs_struct.client_id())?;
+                obj.set_item("issuer", rs_struct.issuer())?;
+                Ok(obj.into_py(py))
+            }
+        }
+    }
+}
```

Comment:
As written, this is a completely opaque type. Unless Rust pyfunctions call `TryFrom`/`TryInto` on it, objects of this type will be returned to Python as a `QcsClientAuthServer` with no methods or accessible fields. The type has a `__new__(cls, Dict[str, Any])` constructor, but that is it.

Just making sure that is what is intended here. Same for below.

Reply:
I'll make [Tokens](https://github.com/rigetti/qcs-api-client-rust/blob/main/qcs-api-client-common/src/configuration/mod.rs#L65-L70) a `py_wrap_data_struct` since the fields are `pub`, but the [`AuthServer`](https://github.com/rigetti/qcs-api-client-rust/blob/main/qcs-api-client-common/src/configuration/settings.rs#L121-L126) fields are private so this needs a proxy `#[pyclass]` I think